promql/engine.go

```diff
@@ -2356,6 +2356,11 @@ loop:
 		} else {
 			histograms = append(histograms, HPoint{H: &histogram.FloatHistogram{}})
 		}
+		if histograms[n].H == nil {
+			// Initialize to non zero to AtFloatHistogram does a copy for sure.
```

Comment:
This doesn't quite parse. The 2nd "to" should be a "so"?

Maybe word it more clearly as: "H must not be nil so that AtFloatHistogram will deep-copy."

Reply:
made a suggestion bellow